external/model-preparation-algorithm/mpa_tasks/extensions/datasets/mpa_cls_dataset.py

```diff
@@ -18,10 +18,16 @@
 @DATASETS.register_module()
 class MPAClsDataset(BaseDataset):
 
-    def __init__(self, ote_dataset=None, labels=None, **kwargs):
+    def __init__(self, ote_dataset=None, labels=None, empty_label=None, multilabel=False,
+                 hierarchical=False, hierarchical_info=None, **kwargs):
```

Comment:
I think the parameter of _init_ seems complicated, what do you think?
Multilabel, hierarchical, and multiclass can be combined into one, and the method of subtracting from kwargs looks better for hierarchical_info. 
Also, it seems good to get a separate parameter for the __init__ of the MPAMultilabelClsDataset.

Reply:
Yes, I agree with your opinion that init seems too complicated. But we if we separate init method, we should handle in task.py.
I think this can be same result compared with current status, just for where we locate complicated statements.
How do you think about that?

Reply:
I still think that even if the task handles it, MPAClsDataset's init should not accept the variables (multilabel, hierarchical, and hierarchical_info...etc) and receive them separately. in the case of multi-class, you can modify it a little so that the variables don't come in in the __init__ place.